common/models.py

```diff
@@ -53,7 +53,8 @@ def _get_conv_tower(self, features):
             if self.dense_net and i != 0:
                 # Dense-net layer input consists of all previous layer outputs.
                 assert previous_output == layer_outputs[-1]
-                inputs = tf.concat(layer_outputs, axis=-1)
+                assert features not in layer_outputs
+                inputs = tf.concat(layer_outputs + [features], axis=-1)
```

Comment:
Is this correct? If this function is supposed to build its own DenseNet block, according to the [paper](https://arxiv.org/pdf/1608.06993.pdf) it should take only feature maps of preceding layers within this block, and not the input itself.

Reply:
If I try to interpret this phrase: "The estimator architecture can be enhanced with DenseNet connections [21]. The inputs to every convolutional layer are the output of and the input to its previous layer", I have to assume "input to the previous layer" includes the feature input of the first layer as well.

Reply:
Densenet paper also quotes the equation "x` = H`([x0, x1, . . . , x`−1]), " where x0 is definitely the input, otherwise there is no definition for how to compute x0. Figure 1 also shows x0 as the input to the block.

Reply:
I think it depends on how you contain the DenseNet block. Perhaps by "input" they meant inputs that were outputs of layers internal to the block? Either way, I can't see this making a huge difference, but it might be safer to stick with the clearer definition of the original paper.

Reply:
The clearer definition shows the input to this block being provided to the other layers.

Reply:
You're right, and it makes more sense this way too. Thanks for checking it over again, my mistake.